annotation-fulltext/src/main/java/eu/europeana/annotation/fulltext/subtitles/SubtitleHandler.java

```diff
@@ -1,52 +1,64 @@
 package eu.europeana.annotation.fulltext.subtitles;
 
+import java.io.BufferedReader;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
 
 import com.dotsub.converter.exception.FileFormatException;
 import com.dotsub.converter.importer.SubtitleImportHandler;
-import com.dotsub.converter.importer.impl.DfxpImportHandler;
-import com.dotsub.converter.importer.impl.QtTextImportHandler;
-import com.dotsub.converter.importer.impl.WebVttImportHandler;
 import com.dotsub.converter.model.Configuration;
 import com.dotsub.converter.model.SubtitleItem;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
 
+import eu.europeana.annotation.config.SubtitleFormat;
+import eu.europeana.annotation.config.SubtitleFormats;
+
+@Component("subtitleHandler")
 public class SubtitleHandler {
 	
-	private static final String INTERNET_MEDIA_TYPE_VTT_FORMAT = "text/vtt";
-	private static final String INTERNET_MEDIA_TYPE_DFXP_FORMAT = "application/ttml+xml";
-	private static final String INTERNET_MEDIA_TYPE_QT_FORMAT = "video/quicktime";
-	
-    private static final SubtitleImportHandler QT_HANDLER = new QtTextImportHandler();
-    private static final SubtitleImportHandler DFXP_HANDLER = new DfxpImportHandler();
-    private static final SubtitleImportHandler VTT_HANDLER = new WebVttImportHandler();
+	public SubtitleHandler(String subtitleFormatsXMLConfigFile, XmlMapper xmlJacksonMapper) throws IOException {
```

Comment:
we will not have more config files, please use a constant for the file name. Also the object mapper is used once during the initialization, we don't need any reference outside the parsing/initialization method

Reply:
The file name is passed as an annotation.property constant in the annotation-web-context.xml file as a constructor argument for the SubtitleHandler class. It is also the case for the object mapper that is declared as a bean in the annotation-web-context.xml file and passed as a constructor argument reference to the SubtitleHandler class.

Reply:
all these are not needed